File: doacao_app_java/src/main/java/exceptions/CustomException.java
package exceptions;

public class CustomException extends Exception {
    private String safeMessage;

    public CustomException(String safeMessage, Throwable cause) {
        super(safeMessage, cause);
        this.safeMessage = safeMessage;
        System.err.println("Erro interno: " + cause.getMessage());
    }

    public CustomException(String safeMessage) {
        super(safeMessage);
        this.safeMessage = safeMessage;
    }

    @Override
    public String getMessage() {
        return safeMessage;
    }

    @Override
    public String toString() {
        return "CustomException: " + safeMessage;
    }
}

File: doacao_animal_python/exceptions/custom_exception.py
class CustomException(Exception):
    def __init__(self, safe_message: str, cause: Exception = None):
        super().__init__(safe_message)
        if cause:
            print(f"Erro interno: {repr(cause)}")
        
    def __str__(self):
        return f"CustomException: {self.args[0]}"